tests/Microsoft.Identity.Test.Integration.netfx/HeadlessTests/ClientCredentialsTests.cs

```diff
@@ -0,0 +1,398 @@
+using System.Diagnostics;
+using System.Globalization;
+using System.IdentityModel.Tokens.Jwt;
+using System.Linq;
+using System.Security.Cryptography;
+using System.Security.Cryptography.X509Certificates;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Identity.Client;
+using Microsoft.Identity.Client.Internal;
+using Microsoft.Identity.Test.Common;
+using Microsoft.Identity.Test.Common.Core.Helpers;
+using Microsoft.Identity.Test.Integration.Infrastructure;
+using Microsoft.Identity.Test.Integration.net45.Infrastructure;
+using Microsoft.Identity.Test.Integration.NetFx.Infrastructure;
+using Microsoft.Identity.Test.LabInfrastructure;
+using Microsoft.Identity.Test.Unit;
+using Microsoft.IdentityModel.JsonWebTokens;
+using Microsoft.IdentityModel.Tokens;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Identity.Test.Integration.HeadlessTests
+{
+
+    [TestClass]
+    public class ClientCredentialsTests
+    {
+        private static readonly string[] s_scopes = { "User.Read" };
+        private static readonly string[] s_keyvaultScope = { "https://vault.azure.net/.default" };        
+        private const string PublicCloudConfidentialClientID = "16dab2ba-145d-4b1b-8569-bf4b9aed4dc8";
+        private const string PublicCloudTestAuthority = "https://login.windows.net/72f988bf-86f1-41af-91ab-2d7cd011db47";
+
+
+        private enum CredentialType { 
+            Cert, 
+            Secret, 
+            ClientAssertion_Wilson, 
+            ClientAssertion_Manual,
+            ClientClaims_MergeClaims,
+            ClientClaims_ExtraClaims
+        };
+
+        [TestInitialize]
+        public void TestInitialize()
+        {
+            TestCommon.ResetInternalStaticCaches();          
+        }
+    
+        [TestMethod]
+        [DataRow(Cloud.Public, RunOn.NetFx | RunOn.NetCore)]
+        [DataRow(Cloud.Adfs, RunOn.NetCore)]
+        [DataRow(Cloud.PPE, RunOn.NetFx)]
+         //[DataRow(Cloud.Arlington)] - cert not setup
+        public async Task WithCertificate_TestAsync(Cloud cloud, RunOn runOn)
+        {
+            runOn.AssertFramework();
+            await RunClientCredsAsync(cloud, CredentialType.Cert).ConfigureAwait(false);
+        }
+
+        [TestMethod]
+        [DataRow(Cloud.Public, RunOn.NetFx | RunOn.NetCore)]
+        [DataRow(Cloud.Adfs, RunOn.NetFx)]        
+        [DataRow(Cloud.Arlington, RunOn.NetCore)]
+        //[DataRow(Cloud.PPE)] - secret not setup
+        public async Task WithSecret_TestAsync(Cloud cloud, RunOn runOn)
+        {
+            runOn.AssertFramework();
+            await RunClientCredsAsync(cloud, CredentialType.Secret).ConfigureAwait(false);
+        }
+
+        [TestMethod]
+        [DataRow(Cloud.Public, RunOn.NetFx | RunOn.NetCore)]
+        [DataRow(Cloud.Adfs, RunOn.NetFx)]
+        [DataRow(Cloud.PPE, RunOn.NetCore)]
+        // [DataRow(Cloud.Arlington)] - cert not setup
```

Comment:
Probably not needed. i think we will be ok with just a client secret test for arlington. Maybe a client cert also.